### PR TITLE
feat: add bigframes.options.compute.maximum_bytes_billed option that sets maximum bytes billed on query jobs

### PR DESCRIPTION
**Pull request by orrbradford**
-implement context manager for global options
-maximum_bytes_billed only applies to query jobs. This limitation will be set per query. Operations that trigger multiple jobs may result in total usage beyond this setting

### CODE REVIEW

File: third_party/bigframes_vendored/pandas/_config/config.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import contextlib
```
Comment (tswast) [inline third_party/bigframes_vendored/pandas/_config/config.py:1]: Needs license header.

Comment (orrbradford) [reply third_party/bigframes_vendored/pandas/_config/config.py:2]: Done. Add a reference to the pandas file similar to other files in the parent directory. Let me know if this works

**Comment by orrbradford** [comment]
kokoro rerun

File: bigframes/pandas/__init__.py [hunk 2]
```diff
@@ -175,6 +175,13 @@ def merge(
     )
 
 
+options = config.options
```
Comment (tswast) [inline bigframes/pandas/__init__.py:178]: This already exists at the bottom of this file. See: https://github.com/googleapis/python-bigquery-dataframes/blob/ac44ccd3936cdb28755d2bbe16377d489f08d5e5/bigframes/pandas/__init__.py#L462

File: bigframes/pandas/__init__.py [hunk 3]
```diff
@@ -175,6 +175,13 @@ def merge(
...
+"""Global :class:`~bigframes._config.Options` to configure BigQuery DataFrames."""
+
+option_context = config.option_context
```
Comment (tswast) [inline bigframes/pandas/__init__.py:181]: Need to add this to `__all__`: https://github.com/googleapis/python-bigquery-dataframes/blob/ac44ccd3936cdb28755d2bbe16377d489f08d5e5/bigframes/pandas/__init__.py#L472

Would also be good to defined below near the `options` definition.

File: bigframes/pandas/__init__.py (hunk 2 above)
Comment (orrbradford) [reply bigframes/pandas/__init__.py]: Done

File: bigframes/pandas/__init__.py (hunk 3 above)
Comment (orrbradford) [reply bigframes/pandas/__init__.py]: Done

File: bigframes/_config/compute_options.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+# limitations under the License.
+
+"""Options for displaying objects."""
+
+import dataclasses
+from typing import Optional
+
+
+@dataclasses.dataclass
+class ComputeOptions:
```
Comment (tswast) [inline bigframes/_config/compute_options.py:22]: Could you add an "autoclass" for this file.

Need to update the following files for this to actually get documented:

* https://github.com/googleapis/python-bigquery-dataframes/blob/main/docs/reference/bigframes/options.rst
* https://github.com/googleapis/python-bigquery-dataframes/blob/main/docs/templates/toc.yml

**Review by tswast — COMMENTED** [review]
Looking good, just one thing I forgot about so that we get this in the docs.

File: bigframes/_config/compute_options.py (hunk 4 above)
Comment (orrbradford) [reply bigframes/_config/compute_options.py:22]: Done

**Review by tswast — APPROVED** [review]

**Pull request merged**